test/functional/p2p_sendtxrcncl.py

```diff
@@ -162,15 +166,17 @@ def run_test(self):
         self.log.info('SENDTXRCNCL if block-relay-only triggers a disconnect')
         peer = self.nodes[0].add_outbound_p2p_connection(
             PeerNoVerack(), wait_for_verack=False, p2p_idx=3, connection_type="block-relay-only")
-        peer.send_message(create_sendtxrcncl_msg(initiator=False))
-        peer.wait_for_disconnect()
+        with self.nodes[0].assert_debug_log(["we indicated no tx relay; disconnecting"]):
+            peer.send_message(create_sendtxrcncl_msg(initiator=False))
+            peer.wait_for_disconnect()
 
         self.log.info('SENDTXRCNCL with initiator=1 and responder=0 from outbound triggers a disconnect')
         sendtxrcncl_wrong_role = create_sendtxrcncl_msg(initiator=True)
         peer = self.nodes[0].add_outbound_p2p_connection(
             P2PInterface(), wait_for_verack=False, p2p_idx=4, connection_type="outbound-full-relay")
-        peer.send_message(sendtxrcncl_wrong_role)
-        peer.wait_for_disconnect()
+        with self.nodes[0].assert_debug_log(["sendtxrcncl received after verack"]):
```

Comment:
I don't think this is the log message we should expect here. I think should be `txreconciliation protocol violation`
Does it pass because this log message happened earlier?

Reply:
Animo

Reply:
`P2PInterface` sends a verack on the version message, so this will disconnect for the verack, not for another reason. Looks like another bug in the test?